sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+from the OS to the memory pool to provide the required memory.
+
+When an asynchornous free command is executed it will free up memory in the 
+memory pool to be used for a later asynchonrous malloc command. Memory is not
+freed back to the OS at this point.
+
+When a SYCL queue which enqueued asynchronous malloc commands is synchronized
+with, i.e. when `queue::wait` oo `queue::wait_and_throw` is called, the SYCL RT\
+will check the current total memory allocated after to the memory pool against
+the memory pool's deallocation threshold. If the total memory allocated islarger
+than than the threshold then the memory will be freed back to the OS, otherwise
+it will not.
+
+[_Note:_ An implementation should maintain memory allocated to a memory pool to
+the size specified by the deallocation threshold when possible, however, an
+implementation is permitted to release this memory back to the OS if it is
+required by another memory pool or an application in another process.
+_{endnote}_]
+
+
+== Implementation notes
+
+It is expected that for L0 this extension will be implemented within the L0
+adapter, by reserving allocations for the memory pool and opportunistically
+re-using the memory allocated based on the command lists being enqueued to the
+L0 driver.
+
+It is expected that for CUDA this extension will be implemented by mapping onto
+the CUDA stream-ordered allocator feature.
+
+
+== Issues
+
+. Should we allow mixing asynchornous and synchornous memory commands?
++
+--
+*UNRESOLVED*: CUDA allows memory allocated with the asynchornous malloc command
+to be freed with the regular sycnhronous free command, should we extend this
+capability to SYCL?
+--
+
+. Should we allow freeing memory with a different queue?
++
+--
+*UNRESOLVED*: Should we allow a memory allocation allocated with an asynchronous
+malloc command from one queue to be freed by an asynchronous free command from
+another queue?
+--
+
+. How should we treat an asynchronous malloc command with a memory pool that is
+not associated with that queue?
++
+--
+*UNRESOLVED*: If an asynchornous malloc command is enqueued with a memory pool
+that is not associated with the queue that the command is enqueued from, should
+this result in an error?
+--
+
+. Should we have a default memory pool for `usm::shared`?
++
+--
+*UNRESOLVED*: Currently the proposed API means that there cannot be default
+memory pool for allocations of allocation kind `usm::shared`, and therefore a
+user must create their own explicit memory pool to do so. Is this reasonable or
+should we extend the API to include a default memory pool for allocations of
+allocation kind `usm::shared`?
+--
+
+. Should we allow setting a new threshold that is lower?
++
+--
+*UNRESOLVED*: Currently setting a new deallocation threshold is only permitted
+if it increases the size of the threshold, however, we may want to also allow
+setting a new lower threshold. This would work by by not immediately freeing any
```

Comment:
```suggestion
setting a new lower threshold. This would work by not immediately freeing any
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.